client/dashboard/domains/name-servers/use-upsell-nudge.ts

```diff
@@ -0,0 +1,21 @@
+import { Domain } from '../../data/domain';
+
+/**
+ * Hook to determine if the upsell nudge should be shown for a domain
+ * @param domain The domain to check
+ * @returns boolean indicating if the upsell nudge should be shown
+ */
+export const useUpsellNudge = ( domain: Domain ): boolean => {
+	if (
+		! domain ||
```

Comment:
By TS definition, this case won't happen

Reply:
True. My focus was on the existing (current component) if statement. I tried not to miss anything, but I completely lost that part of the context. I'll remove it.